qutip/bloch.py

```diff
@@ -799,52 +785,52 @@ def plot_vectors(self):
 
     def plot_points(self):
         # -X and Y data are switched for plotting purposes
-        for k in range(len(self.points)):
-            num = len(self.points[k][0])
-            dist = [sqrt(self.points[k][0][j] ** 2 +
-                         self.points[k][1][j] ** 2 +
-                         self.points[k][2][j] ** 2) for j in range(num)]
-            if any(abs(dist - dist[0]) / dist[0] > 1e-12):
-                # combine arrays so that they can be sorted together
-                zipped = list(zip(dist, range(num)))
-                zipped.sort()  # sort rates from lowest to highest
-                dist, indperm = zip(*zipped)
-                indperm = array(indperm)
+        for k, points in enumerate(self.points):
+            points = np.asarray(points)
+            num_points = points.shape[1]
+
+            dist = np.linalg.norm(points, axis=0)
+            if not np.allclose(dist, dist[0], rtol=1e-12):
+                indperm = np.argsort(dist)
+                points = points[:, indperm]
             else:
-                indperm = arange(num)
-            if self.point_style[k] == 's':
-
-                self.axes.scatter(
-                    real(self.points[k][1][indperm]),
-                    - real(self.points[k][0][indperm]),
-                    real(self.points[k][2][indperm]),
-                    s=self.point_size[mod(k, len(self.point_size))],
-                    alpha=self.point_alpha[k],
-                    edgecolor=None,
-                    zdir='z',
-                    color=self.point_color[k][0],
-                    marker=self.point_marker[mod(k, len(self.point_marker))])
-
-            elif self.point_style[k] == 'm':
-                pnt_colors = array(self.point_color *
-                                   int(ceil(num / float(
-                                       len(self.point_color)))))
-
-                marker = self.point_marker[mod(k, len(self.point_marker))]
-                s = self.point_size[mod(k, len(self.point_size))]
-                self.axes.scatter(real(self.points[k][1][indperm]),
-                                  -real(self.points[k][0][indperm]),
-                                  real(self.points[k][2][indperm]),
-                                  s=s, alpha=self.point_alpha[k],
-                                  edgecolor=None, zdir='z',
-                                  color=self.point_color[k][0], marker=marker)
+                indperm = np.arange(num_points)
+
+            s = self.point_size[mod(k, len(self.point_size))]
+            marker = self.point_marker[mod(k, len(self.point_marker))]
+            user_color = self.point_color[k]
+            style = self.point_style[k]
+            if user_color is not None:
+                color = user_color
+            elif user_color is None and self.point_style[k] in ['s', 'l']:
```

Comment:
```suggestion
            elif self.point_style[k] in ['s', 'l']:
```
`user_color` is always `None` in the `elif`.